scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1386,186 @@ cdef class Rotation:
+              `axes`, which corresponds to a single rotation with `W` axes
+            - array_like with shape (N, W) where each `angle[i]`
+              corresponds to a sequence of Davenport angles describing a 
+              single rotation
+
+        extrinsic : boolean
+            If True, sequence will be extrinsic. If False, sequence will be
+            treated as intrinsic.
+        degrees : bool, optional
+            If True, then the given angles are assumed to be in degrees. 
+            Default is False.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotation represented by the sequence of
+            rotations around given axes with given angles.
+
+        References
+        ----------
+        .. [1] https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations
+        .. [2] Shuster, Malcolm & Markley, Landis. (2003). Generalization of
+               the Euler Angles. Journal of the Astronautical Sciences. 51. 123-132. 10.1007/BF03546304.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Davenport angles are a generalization of Euler angles, when we use the
+        canonical basis axes:
+
+        >>> ex = [1, 0, 0]
+        >>> ey = [0, 1, 0]
+        >>> ez = [0, 0, 1]
+
+        Initialize a single rotation with a given axis sequence:
+
+        >>> axes = [ez, ey, ex]
+        >>> r = R.from_davenport(axes, [90, 0, 0], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        It is equivalent to Euler angles in this case:
+
+        >>> r.as_euler('zyx', degrees=True)
+        array([90.,  0., -0.])
+
+        Initialize multiple rotations in one object:
+
+        >>> r = R.from_davenport(axes, [[90, 45, 30], [35, 45, 90]], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (2, 4)
+
+        Non-canonical axes are possible, and they do not need to be normalized,
+        as long as consecutive axes are orthogonal:
+
+        >>> e1 = [2, 0, 0]
+        >>> e2 = [0, 1, 0]
+        >>> e3 = [1, 0, 1]
+        >>> axes = [e1, e2, e3]
+        >>> r = R.from_davenport(axes, [90, 45, 30], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        """
+        axes = np.asarray(axes)
+        if axes.ndim == 1:
+            axes = axes.reshape([1, 3])
+
+        num_axes = len(axes)
+
+        for axis in axes:
+            if len(axis) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected up to 3 axes, got {}".format(num_axes))
+
+        # normalize axes
+        norm = np.repeat(np.linalg.norm(axes, axis=1), 3)
+        axes = axes / norm.reshape(num_axes, 3)
+
+        if num_axes > 1 and abs(np.dot(axes[0], axes[1])) >= 1e-7:
```

Comment:
Any reason to choose this particular threshold??

Reply:
Not really, just some threshold sufficiently close to zero. Should I change it to something else?

Reply:
It's always tricky to handle such situations.

As far as I can see ,`as_davenport` works formally for any axes (because it's implemented in a robust way using `atan2` etc.). 

So "close to orthogonal" axes should give a "almost correct" and meaningful result without failure.

In this case I have nothing against with threshold.

Does it make sense?

Reply:
Yeah, I see what you mean! In any case I think the most important would be to have some kind of reasonable threshold so that someone tried to use orthogonal axes, they would get this error and understand that it's not really possible with Davenport angles (similarly to how Euler axes cannot be consecutive).

I'd leave this like it is, if that's ok for you!